lib/iris/tests/unit/analysis/trajectory/test_interpolate.py

```diff
@@ -46,42 +48,52 @@ class TestNearest:
     # That has its own test, so we don't test the basic calculation
     # exhaustively here.  Instead we check the way it handles the source and
     # result cubes (especially coordinates).
-    @pytest.fixture(autouse=True)
-    def setup(self):
+    @pytest.fixture
+    def src_cube(self):
         cube = iris.tests.stock.simple_3d()
         # Actually, this cube *isn't* terribly realistic, as the lat+lon coords
         # have integer type, which in this case produces some peculiar results.
         # Let's fix that (and not bother to test the peculiar behaviour).
         for coord_name in ("longitude", "latitude"):
             coord = cube.coord(coord_name)
             coord.points = coord.points.astype(float)
-        self.test_cube = cube
+        return cube
+
+    @pytest.fixture
+    def single_point(self, src_cube):
         # Define coordinates for a single-point testcase.
         y_val, x_val = 0, -90
-        # Work out cube indices of the testpoint.
-        self.single_point_iy = np.where(
-            cube.coord("latitude").points == y_val
-        )[0][0]
-        self.single_point_ix = np.where(
-            cube.coord("longitude").points == x_val
-        )[0][0]
+
         # Use slightly-different values to test nearest-neighbour operation.
-        self.single_sample_point = [
+        sample_point = [
             ("latitude", [y_val + 19.23]),
             ("longitude", [x_val - 17.54]),
         ]
 
+        # Work out cube indices of the testpoint.
+        single_point_iy = np.where(src_cube.coord("latitude").points == y_val)[
+            0
+        ][0]
+        single_point_ix = np.where(
+            src_cube.coord("longitude").points == x_val
+        )[0][0]
+
+        point = namedtuple("point", "ix iy sample_point")
+        return point(single_point_ix, single_point_iy, sample_point)
```

Comment:
I have used a namedtuple here as this fixture returns multiple values but I want to avoid the boiler plate of having to unpack a tuple each time